tasmota/tasmota_support/support.ino

```diff
@@ -590,6 +590,33 @@ char* Trim(char* p) {
   return p;
 }
 
+size_t HexStringToBinaryString(char** source) {
+  size_t str_len = strlen(*source+2);
+  char *s = *source+2;
+  char *new_str = (char *)malloc(str_len/2);
+  for (int i = 0; i < str_len; i++) {
+        unsigned char val;
+        char cur = s[i];
+        if (cur >= 97) {
+            val = cur - 97 + 10;
+        } else if (cur >= 65) {
+            val = cur - 65 + 10;
+        } else {
+            val = cur - 48;
+        }
+        /* even characters are the first half, odd characters the second half
+         * of the current output byte */
+        if (i%2 == 0) {
```

Comment:
Maybe a bit picky, but I prefer `i & 1` which is just a bit mask, instead of a full-blown division. (Maybe the compiler replaces it already)